pytlu/tlu.py

```diff
@@ -36,14 +39,91 @@ def handle_sig(signum, frame):
     stop_run = True
 
 
+def parse_arguments(eudaq=False):
+    def th_type(x):
+        if int(x) > 31 or int(x) < 0:
+            raise argparse.ArgumentTypeError("Threshold is 0 to 31")
+        return int(x)
+
+    parser = argparse.ArgumentParser(usage="pytlu -ie CH0 -oe CH0",
+                                     description='TLU DAQ\n TX_STATE: 0= DISABLED 1=WAIT 2=TRIGGERED (wait for busy HIGH) 4=READ_TRIG (wait for busy LOW) LBS is CH0', formatter_class=argparse.RawTextHelpFormatter)
+
+    parser.add_argument('-ie', '--input_enable', nargs='+', type=str, choices=input_ch, default=[],
+                        help='Enable input channels. Allowed values are ' + ', '.join(input_ch), metavar='CHx')
+    parser.add_argument('-oe', '--output_enable', nargs='+', type=str, choices=output_ch, default=[],
+                        help='Enable ouput channels. CHx and LEM0x are exclusive. Allowed values are ' + ', '.join(output_ch), metavar='CHx/LEMOx')
+    parser.add_argument('-th', '--threshold', type=th_type, default=0,
+                        help="Digital threshold for input (in units of 1.5625ns). Default=0", metavar='0...31')
+    parser.add_argument('-b', '--n_bits_trig_id', type=th_type, default=16,
+                        help="Number of bits for trigger ID. Should correspond to TLU_TRIGGER_MAX_CLOCK_CYCLES - 1 which is set for TLU module. Default=0", metavar='0...31')
+    parser.add_argument('-ds', '--coincidence_window', type=th_type, default=31,
```

Comment:
Maximum is 32?

Reply:
31 is the maximum, this register is 5 bit large and 0 means disabled.